fvm/fvm_test.go

```diff
@@ -1085,7 +1085,22 @@ func TestSettingExecutionWeights(t *testing.T) {
 			require.NoError(t, err)
 
 			tx := fvm.Transaction(txBody, programs.NextTxIndexForTestingOnly())
-			err = vm.RunV2(ctx, tx, view)
+
+			defaultMeter := meter.DefaultParameters()
+			meterParams, err := fvm.GetEnvironmentMeterParameters(
```

Comment:
If you move 
```
meterParams := meter.DefaultParameters().
		WithComputationLimit(uint(proc.ComputationLimit(ctx))).
		WithMemoryLimit(proc.MemoryLimit(ctx))
```
from fvm/fvm.go to inbetween these to lines, the `if ctx.MeterParameters.TotalMemoryLimit() != math.MaxUint64` isnt needed anymore. Also the `RunV2` can directly use `ctx.MeterParameters` without any further adjustment.